messages.go

```diff
@@ -182,3 +182,157 @@ func NewMsg(msgType string, parent ComposedMsg) (ComposedMsg, error) {
 
 	return msg, nil
 }
+
+// Publish creates a new ComposedMsg and sends it back to the return identities over the
+// IOPub channel
+func (receipt *msgReceipt) Publish(msgType string, content interface{}) error {
+	msg, err := NewMsg(msgType, receipt.Msg)
+
+	if err != nil {
+		return err
+	}
+
+	msg.Content = content
+	return receipt.SendResponse(receipt.Sockets.IOPubSocket, msg)
+}
+
+// Reply creates a new ComposedMsg and sends it back to the return identities over the
+// Shell channel
+func (receipt *msgReceipt) Reply(msgType string, content interface{}) error {
+	msg, err := NewMsg(msgType, receipt.Msg)
+
+	if err != nil {
+		return err
+	}
+
+	msg.Content = content
+	return receipt.SendResponse(receipt.Sockets.ShellSocket, msg)
+}
+
+// MIMEDataBundle holds data that can be presented in multiple formats. The keys are MIME types
+// and the values are the data formatted with respect to it's MIME type. All bundle should contain
+// at least a "text/plain" representation with a string value.
+type MIMEDataBundle map[string]interface{}
```

Comment:
Is there a reason to define this type specifically, or could we just return and utilize `map[string]interface{}` throughout?

Reply:
The reason for the alias is to help make the code that uses it a bit more self documenting. Otherwise it is a little unclear what a `map` would hold for output.

Reply:
I'll leave this one up to you.  If you think it adds value, I'm ok with it.